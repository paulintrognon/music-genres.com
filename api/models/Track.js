const Sequelize = require('sequelize');
const sequelize = require('../db.js').sequelize;

const MusicGenre = require('./MusicGenre.js');

const Track = sequelize.define('track', {
  url: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [5, 200],
  },
});

Track.belongsTo(MusicGenre);

Track.sync();

module.exports = Track;
