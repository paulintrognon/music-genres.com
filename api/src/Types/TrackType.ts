export type TrackType = {
  id: number
  title: string
  playerName: 'youtube'
  playerTrackId: string
}
