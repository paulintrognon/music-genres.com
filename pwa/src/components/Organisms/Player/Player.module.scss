.container {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(57, 63, 127,.9);
  z-index: 10;
  display: flex;
  justify-content: center;
  align-items: center;
}

.closeContainer {
  text-align: right;
  margin-bottom: 0.5rem;
}

.closeIcon {
  position: relative;
}
