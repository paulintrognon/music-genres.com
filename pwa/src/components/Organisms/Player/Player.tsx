import { useDispatch, useSelector } from 'react-redux'
import { animated, useTransition } from 'react-spring'
import { close } from '../../../redux/player/player.slice'
import { RootState } from '../../../redux/store'
import CrossIcon from '../../Atoms/Icons/CrossIcon/CrossIcon'
import classes from './Player.module.scss'

const Player: React.FC = () => {
  const dispatch = useDispatch()
  const { track } = useSelector((state: RootState) => state.player)
  const transition = useTransition(track, null, {
    from: { opacity: '0' },
    enter: { opacity: '1' },
    leave: { opacity: '0' },
    config: { duration: 200 },
  })

  return (
    <>
      {transition.map(
        ({ item, key, props }) =>
          item && (
            <animated.div style={props} key={key} className={classes.container}>
              <div>
                <div className={classes.closeContainer}>
                  <button onClick={() => dispatch(close())} className={classes.closeIcon}>
                    <CrossIcon />
                  </button>
                </div>
                <iframe
                  width="646"
                  height="405"
                  frameBorder="0"
                  title="Youtube Player"
                  src={`https://www.youtube.com/embed/${track?.playerTrackId}?autoplay=1&start=20&origin=https://www.music-genres.com`}
                />
              </div>
            </animated.div>
          )
      )}
    </>
  )
}
export default Player
