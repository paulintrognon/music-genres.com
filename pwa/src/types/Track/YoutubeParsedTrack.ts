export type YoutubeParsedTrack = {
  playerName: 'youtube'
  playerTrackId: string
  title: string
  description: string
  tags: string[]
  owner: { name: string; id: string }
}
