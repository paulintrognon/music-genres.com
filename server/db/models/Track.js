const Sequelize = require('sequelize');
const sequelize = require('../db.js').sequelize;

const Track = sequelize.define('track', {
  serviceName: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [3, 30],
  },
  serviceTrackId: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [1, 255],
  },
  /*title: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [1, 100],
  },*/
  upvotes: {
    type: Sequelize.INTEGER,
    allowNull: false,
    defaultValue: 0,
  },
});

module.exports = Track;
