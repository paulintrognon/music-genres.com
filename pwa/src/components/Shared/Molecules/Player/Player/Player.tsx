import { TrackType } from '../../../../../types/Track/TrackType'
import Fader from '../../../Animations/Fader/Fader'
import CrossIcon from '../../../Atoms/Icons/CrossIcon/CrossIcon'
import classes from './Player.module.scss'

interface Props {
  track?: TrackType
  onClose: () => void
}
const Player: React.FC<Props> = ({ track, onClose }) => {
  return (
    <Fader isVisible={Boolean(track)}>
      <div className={classes.container}>
        <div>
          <div className={classes.closeContainer}>
            <button onClick={onClose} className={classes.closeIcon} aria-label="Close">
              <CrossIcon />
            </button>
          </div>
          <iframe
            width="646"
            height="405"
            frameBorder="0"
            title="Youtube Player"
            src={`https://www.youtube.com/embed/${track?.playerTrackId}?autoplay=1&start=20&origin=https://www.music-genres.com`}
          />
        </div>
      </div>
    </Fader>
  )
}
export default Player
