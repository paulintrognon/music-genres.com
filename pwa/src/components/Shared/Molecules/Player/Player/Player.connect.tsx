import { useDispatch, useSelector } from 'react-redux'
import { close } from '../../../../../redux/player/player.slice'
import { RootState } from '../../../../../redux/store'
import Player from './Player'

const ConnectedPlayer: React.FC = () => {
  const dispatch = useDispatch()
  const { track } = useSelector((state: RootState) => state.player)

  const handleOnClose = (): void => {
    dispatch(close())
  }

  return <Player track={track} onClose={handleOnClose} />
}
export default ConnectedPlayer
