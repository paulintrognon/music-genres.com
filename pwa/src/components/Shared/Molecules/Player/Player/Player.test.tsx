import { shallow } from 'enzyme'
import { TrackType } from '../../../../../types/Track/TrackType'
import Player from './Player'

it('should render the Player with visible = false if no track is provided', () => {
  const wrapper = shallow(<Player onClose={jest.fn()} />)
  expect(wrapper).toMatchSnapshot()
})

it('should render the Player with visible = true if track is provided', () => {
  const track: TrackType = {
    id: 1,
    title: 'Abc 123',
    playerName: 'youtube',
    playerTrackId: 'abc123',
  }
  const wrapper = shallow(<Player track={track} onClose={jest.fn()} />)
  expect(wrapper).toMatchSnapshot()
})

it('should call onClose when clicking on close icon', () => {
  const onClose = jest.fn()
  const wrapper = shallow(<Player onClose={onClose} />)
  wrapper.find('[aria-label="Close"]').simulate('click')
  expect(onClose).toHaveBeenCalledTimes(1)
})
