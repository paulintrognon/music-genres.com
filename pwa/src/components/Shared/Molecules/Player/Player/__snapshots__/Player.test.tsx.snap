// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Player with visible = false if no track is provided 1`] = `
<Fader
  isVisible={false}
>
  <div
    className="container"
  >
    <div>
      <div
        className="closeContainer"
      >
        <button
          aria-label="Close"
          className="closeIcon"
          onClick={[MockFunction]}
        >
          <CrossIcon />
        </button>
      </div>
      <iframe
        frameBorder="0"
        height="405"
        src="https://www.youtube.com/embed/undefined?autoplay=1&start=20&origin=https://www.music-genres.com"
        title="Youtube Player"
        width="646"
      />
    </div>
  </div>
</Fader>
`;

exports[`should render the Player with visible = true if track is provided 1`] = `
<Fader
  isVisible={true}
>
  <div
    className="container"
  >
    <div>
      <div
        className="closeContainer"
      >
        <button
          aria-label="Close"
          className="closeIcon"
          onClick={[MockFunction]}
        >
          <CrossIcon />
        </button>
      </div>
      <iframe
        frameBorder="0"
        height="405"
        src="https://www.youtube.com/embed/abc123?autoplay=1&start=20&origin=https://www.music-genres.com"
        title="Youtube Player"
        width="646"
      />
    </div>
  </div>
</Fader>
`;
